.vib/argo-cd/cypress/cypress/support/commands.js

```diff
@@ -0,0 +1,42 @@
+const COMMAND_DELAY = 2000;
+
+for (const command of ['click']) {
+  Cypress.Commands.overwrite(command, (originalFn, ...args) => {
+    const origVal = originalFn(...args);
+
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(origVal);
+      }, COMMAND_DELAY);
+    });
+  });
+}
+
+Cypress.Commands.add(
+  'login',
+  (username = Cypress.env('username'), password = Cypress.env('password')) => {
+    cy.visit('/login');
+
+    // Checking for UI elements before and after login avoids race-condition errors
+    cy.get('.login__box');
```

Comment:
The issue here is that Cypress fails intermittently when it looks for an element inside the `.login__box` one. This is caused by the way the UI is loaded, which makes Cypress fail if the parent of a searched element is hidden. I decided to look for the parent element to avoid this error.

Having said that, I checked whether using some other element for this "wait" was better but I actually didn't like any of them. I opted for using `.login__box` because it seemed like one of the less change-prone selectors available.